lib/src/rules/give_deprecation_message.dart

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2018, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:linter/src/analyzer.dart';
+
+const _desc = r'Give a deprecation message, via `Deprecated("message")`.';
+
+const _details = r'''
+
+**DO** specify a deprecation message (with migration instructions and/or a
+removal schedule) in the Deprecation constructor.
+
+**BAD:**
+```
+@deprecated
+void oldFunction(arg1, arg2) {}
+```
+
+**GOOD:**
+```
+@Deprecated("""
+[oldFunction] is being deprecated in favor of [newFunction] (with slightly
+different parameters; see [newFunction] for more information). [oldFunction]
+will be removed on or after the 4.0.0 release.
+""")
+void oldFunction(arg1, arg2) {}
+```
+
+''';
+
+class GiveDeprecationMessage extends LintRule implements NodeLintRule {
+  GiveDeprecationMessage()
+      : super(
+            name: 'give_deprecation_message',
```

Comment:
I'm not fond of 'give'. Perhaps something like 'provide' or 'include'?

Reply:
+1. "provide" or "include" sound a little better to me too.

Reply:
Done.